.github/workflows/backend-docs-generation.yml

```diff
@@ -0,0 +1,36 @@
+name: Generate documentation for the backend
+
+on:
+    push:
+        branches:
+            - '**'
+    pull_request:
+        branches:
+            - '**'
+jobs:
+    generate-docs:
+        runs-on: ubuntu-latest
+        steps:
+            - name: Checkout-Repository
+              uses: actions/checkout@v3
+        
+            - uses: actions/setup-java@v2
+              with:
+                distribution: 'temurin'
+                java-version: '17'
+
+            - name: Install asciidoctor and Asciidoctor pdf
+              run: |
+                sudo apt-get update
+                sudo apt-get install -y asciidoctor
+                sudo gem install asciidoctor-pdf --pre
+
+            - name: Generate-pdf-docs
+              run: ./gradlew convertAdocToPdf
+
+            - name: Upload PDF artifact
+              uses: actions/upload-artifact@v4
+              with:
+                    name: documentation.pdf
+                    path: ./build/pdf
```

Comment:
After this upload, one should be capable of downloading and sharing the pdfs. How can we do that?

Reply:
After running the workflow, there is a link in the logs of the 'upload' action which helps you to download the created pdf

Reply:
Very good! Thank you! Then it means that you can document it out somewhere and merge this ticket, isn't it? Thank you 🙏🏿